lib/google_search_data_viewer_web/plugs/put_current_user.ex

```diff
@@ -0,0 +1,17 @@
+defmodule GoogleSearchDataViewerWeb.PutCurrentUserPlug do
+  import Plug.Conn
+
+  def init(opts), do: opts
+
+  def call(conn, _opts) do
+    user_id = get_session(conn, :user_id)
+
+    user = user_id && GoogleSearchDataViewer.Accounts.get_user(user_id)
+
+    if user do
+      assign(conn, :current_user, user)
+    else
+      assign(conn, :current_user, nil)
+    end
```

Comment:
Can we just?

```suggestion
assign(conn, :current_user, user)
```

Reply:
Oh nice, yes we can as ```Plug.get_session``` and ```Repo.get```  both return **nil** and so wont still work correctly using the value in the template. Thanks.